src/sema/types.rs

```diff
@@ -247,21 +248,17 @@ pub fn resolve_fields(delay: ResolveFields, file_no: usize, ns: &mut Namespace)
         check(struct_no, &mut vec![struct_no], ns);
     }
 
-    // Do not attempt to call struct offsets if there are any infinitely recursive structs
-    if !ns.diagnostics.any_errors() {
-        struct_offsets(ns);
-    }
+    // Calculate the offset of each field in all the struct types
+    struct_offsets(ns);
```

Comment:
If a struct has itself as a member, solang will enter an infinite loop when calculating offsets. There should a way to stop the loop.

Take this contract as an example:

```solidity
contract tt {
    struct rr {
        uint a;
        rr b;
    }

    rr s_rr;
    function getThis() public returns (uint) {
        return s_rr.a;
    }
}
```

Reply:
Right. Looks like this goes into an infinite loop when compiling for solana, which was not covered by the tests yet. Fixed.

Reply:
I couldn't find any test case for the recursive struct. I believe it is a good idea to have it.

Reply:
Just look for " has infinite size". There are a few tests in the `tests/contract_tests/`. I've moved one substrate test to solana in this PR.

Reply:
As I said above, this PR moves   substrate test to solana